### PR TITLE
[AST/Sema] Make it possible to use init accessors in inlinable initializers

### PR DESCRIPTION
**Pull request by xedin**
- Make init accessors `internal`
- Require explicit `@usableFromInline` on init accessor when property is used in a designated inlinable initializer
- Improve diagnostics when `@usableFromInline` is applied to a declaration without a name (i.e. accessor/init)

Resolves: rdar://129318806

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test Linux platform

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Comment by xedin** [comment]
@swift-ci please test Windows platform

**Review by tshortli — APPROVED** [review]

File: lib/Sema/ResilienceDiagnostics.cpp [hunk 1]
```diff
@@ -64,6 +64,17 @@ bool TypeChecker::diagnoseInlinableDeclRefAccess(SourceLoc loc,
   auto *DC = where.getDeclContext();
   auto &Context = DC->getASTContext();
 
+if (auto *init = dyn_cast<ConstructorDecl>(DC)) {
+    if (init->isDesignatedInit()) {
```
[the remark this answers is not in the document]
Comment (xedin) [reply lib/Sema/ResilienceDiagnostics.cpp:68]: Weird, thank you!

[the remark this answers is not in the document]
Comment (xedin) [reply lib/Sema/ResilienceDiagnostics.cpp:68]: ✅

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Pull request merged**